### PR TITLE
Improve hostname validation

### PR DESCRIPTION
**Pull request by AlexTugarev**

### CODE REVIEW

File: components/server/src/bitbucket/bitbucket-context-parser.spec.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ import { BitbucketTokenHelper } from "./bitbucket-token-handler";
 const expect = chai.expect;
 import { skipIfEnvVarNotSet } from "@gitpod/gitpod-protocol/lib/util/skip-if";
 
-@suite.only(timeout(10000), skipIfEnvVarNotSet("GITPOD_TEST_TOKEN_BITBUCKET"))
```
Comment (geropl) [inline components/server/src/bitbucket/bitbucket-context-parser.spec.ts:21]: :100:

File: components/server/src/express-util.ts [hunk 2]
```diff
@@ -91,7 +91,14 @@ const looksLikeWorkspaceHostname = (originHostname: URL, gitpodHostName: string)
         return false;
     }
     const url = new GitpodHostUrl(originHostname);
-    return workspaceIDRegex.test(url.workspaceId || '')
+    const workspaceId = url.workspaceId;
+    if (workspaceId) {
+        const hostname = url.url.hostname as string;
+        if (hostname.startsWith(workspaceId)) {
```
Comment (geropl) [inline components/server/src/express-util.ts:97]: Would be nice if we had one place for the workspace URL parsing. But might be out of scope for this PR, as it was here before, too.

**Review by geropl — APPROVED** [review]
Tested, works fine. Also, re-enables the test that failed because of this! :heart:

File: components/server/src/express-util.ts (hunk 2 above)
Comment (AlexTugarev) [reply components/server/src/express-util.ts:97]: `GitpodHostUrl` does the URL parsing. Here we explicitly check, that the workspace-like hostname is not prefixed.
Ah, perhaps you mean to extract the computation of the prefix and move it to `GitpodHostUrl`?  🤷🏻‍♂️

**Pull request merged**